app/stac_api/models.py

```diff
@@ -125,48 +145,137 @@ class Collection(models.Model):
+    def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asset_eo_gsd):
+        '''
+        updates the collection's summaries when assets are updated or raises
+        errors when this fails.
+        :param asset_geoadmin_value: asset's value for geoadmin_variant
+        :param asset_proj_epsg: asset's value for proj:epsg
+        :param asset_eo_gsd: asset's value for asset_eo_gsd
+        For all the given parameters this function checks, if the corresponding
+        parameters of the collection need to be updated. If so, they will be either
+        updated or an error will be raised, if updating fails.
+        '''
+        try:
+            if self.summaries["geoadmin:variant"] is None:
+
+                self.summaries["geoadmin:variant"] = [asset_geoadmin_variant]
+                self.save()
+
+            elif asset_geoadmin_variant not in self.summaries["geoadmin:variant"]:
+                self.summaries["geoadmin:variant"].append(asset_geoadmin_variant)
+                self.save()
 
+            if self.summaries["proj:epsg"] is None:
+                self.summaries["proj:epsg"] = [asset_proj_epsg]
+                self.save()
+
+            elif asset_proj_epsg not in self.summaries["proj:epsg"]:
+                self.summaries["proj:epsg"].append(asset_proj_epsg)
+                self.save()
+
+            if self.summaries["eo:gsd"] is None:
+                self.summaries["eo:gsd"] = [asset_eo_gsd]
+                self.save()
+
+            elif not float_in(asset_eo_gsd, self.summaries["eo:gsd"]):
+                self.summaries["eo:gsd"].append(asset_eo_gsd)
+                self.save()
+
+        except (KeyError, IndexError) as err:
+            logger.error(_(
+                "Error when updating collection's summaries values due to asset update: %s"), err
+                        )
+            raise ValidationError(_(
+                "Error when updating collection's summaries values due to asset update."
+            ))
+
+    def update_extent(self, item_properties_datetime):
+        '''
+        updates the collection's temporal extent when item's are update.
+        :param item_properties_datetime: item's value for properties_datetime.
+        This function checks, if the corresponding parameter of the collection
+        needs to be updated. If so, it will be either updated or an error will
+        be raised, if updating fails.
+        '''
+        try:
+
+            if self.extent["temporal"]["interval"][0][0] is None:
+
+                self.extent["temporal"]["interval"][0][0] = item_properties_datetime
+                self.save()
+
+            elif item_properties_datetime < self.extent["temporal"]["interval"][0][0]:
+
+                self.extent["temporal"]["interval"][0][0] = item_properties_datetime
+                self.save()
+
+            elif self.extent["temporal"]["interval"][0][1] is None:
```

Comment:
Should not it be `if` here ? and same comment about blank lines as above.

Reply:
From my understanding, `elif` is correct, but the whole block is quite nasty to read.

When reaching the line (formerly known as line 245), so far it has been checked, that:
* start_date in the extent's temporal inteval is **not** `None` **and**
* current item instance's datetime value is **not** < extent's temporal interval's start_date, so **only** then:
* check if end_date in extent's temporal interval is `None` and if so,
* set that to the current item's instances datetime value

Hence, it should be an `eflif` and not only an `if` here, from my understanding.